uniffi_bindgen/src/interface/mod.rs

```diff
@@ -119,6 +121,83 @@ impl<'ci> ComponentInterface {
         Ok(ci)
     }
 
+    /// Parse a `ComponentInterface` from a string containing a Rust source file.
+    pub fn from_rust(rs: &str) -> Result<Self> {
+        let mut ci = Self::default();
+        ci.uniffi_version = env!("CARGO_PKG_VERSION").to_string();
+        // There's some lifetime thing with the errors returned from parsing
+        // that my own lifetime is too short to worry about figuring out; unwrap and move on.
+        let file_src = syn::parse_file(rs).expect("Failed to parse Rust code");
+        // First, let's see what sort of style of file we're dealing with.
+        // It might be `include_scaffolding!` or it might be `#[declare_interface]`.
```

Comment:
This is currently a bit weird. I'm trying to support parsing both the new `#[declare_interface]` syntax proposed in this PR, and also the existing `include_scaffolding!()` syntax that is offered by the current release version of UniFFI. That's probably not worthwhile except to support short-term experiments with the parsing logic here.

Reply:
agreed!